File: server/src/domain/dtos/messages/create-message.dto.ts
export class CreateMessageDto {
    private constructor(
        public readonly sender: string,
        public readonly content: string,
        public readonly chat: string,
        public readonly image: string,
    ) { }

    static create(props: { [key: string]: string }): [string?, CreateMessageDto?] {
        const { sender, content, chat, image } = props;

        if (!sender) return ['Missing sender'];
        if (!content && !image) return ['Missing content'];
        if (!chat) return ['Missing chat'];
        if (image !== undefined && typeof (image) !== 'string') return ['Missing must be a string'];

        return [undefined, new CreateMessageDto(sender, content, chat, image)];
    }
}

File: ServerNET/Application/DTOs/Messages/CreateMessageDto.cs
namespace Application.DTOs.Messages;

public class CreateMessageDto
{
    public string Sender { get; set; } = null!;
    public string Content { get; set; } = null!;
    public string Chat { get; set; } = null!;
    public string Image { get; set; } = null!;
    public bool IsSeen { get; set; }
}

